api/src/main/java/jakarta/data/repository/Select.java

```diff
@@ -0,0 +1,135 @@
+     * The result type that is used in the method return type must be the
+     * type of the entity attribute.</p>
+     *
+     * <p>For example, to return only the {@code price} attribute of the
+     * {@code Car} entity that has the supplied {@code vin} attribute value,</p>
+     *
+     * <pre>
+     * &#64;Repository
+     * public interface Cars extends BasicRepository&lt;Car, String&gt; {
+     *     &#64;Find
+     *     &#64;Select(_Car.PRICE)
+     *     Optional&lt;Float&gt; getPrice(@By(_Car.VIN) String vehicleIdNum);
+     * }
+     * </pre>
+     *
+     * <h2>Method that returns Java Records</h2>
+     *
+     * <p>A repository method can return a subset of entity attributes per result
+     * by having the result type be a Java record. The {@code Select} annotation
+     * can be used in the following ways to accommodate this.</p>
+     *
+     * <h3>Annotating a Repository Method</h3>
+     *
+     * <p>Place the {@code Select} annotation on a repository find method and
+     * assign the annotation value to be the names of multiple entity attributes,
+     * corresponding to the order and types of the components of the Java record
+     * that is used for the result type.</p>
+     *
+     * <p>For example, to return only the {@code model}, {@code make}, and
+     * {@code year} attributes of a {@code Car} entity that has the supplied
+     * {@code vin} attribute value,</p>
+     *
+     * <pre>
+     * &#64;Repository
+     * public interface Cars extends BasicRepository&lt;Car, String&gt; {
+     *     record ModelInfo(String model,
+     *                      String manufacturer,
+     *                      int designYear) {}
+     *
+     *     &#64;Find
+     *     &#64;Select({_Car.MODEL, _Car.MAKE, _Car.YEAR})
+     *     Optional&lt;ModelInfo&gt; getModelInfo(@By(_Car.VIN) String vehicleIdNum);
+     * }
+     * </pre>
+     *
+     * <h3>Annotating a Record Component</h3>
+     *
+     * <p>Place the {@code Select} annotation on each record component of the
+     * Java record that is used as the result type of the repository method.
+     * Assign the annotation value to be the name of an entity attribute that
+     * has the same type as the record component.</p>
+     *
+     * <pre>
+     * &#64;Repository
+     * public interface Cars extends BasicRepository&lt;Car, String&gt; {
+     *     record ModelInfo(&#64;Select(_Car.MODEL) String model,
+     *                      &#64;Select(_Car.MAKE) String manufacturer,
+     *                      &#64;Select(_Car.YEAR) int designYear) {}
+     *
+     *     &#64;Find
+     *     Optional&lt;ModelInfo&gt; getModelInfo(@By(_Car.VIN) String vehicleIdNum);
+     * }
+     * </pre>
+     *
+     * <p>For more concise code, the {@code Select} annotation can be omitted from
+     * record components that have the same name as the entity attribute name,
+     * such as {@code model} in the above example.</p>
```

Comment:
Will this also require the `--parameters` compiler option? If so should we point that out here?

Reply:
No, unlike method parameter names, which require `--parameters`, record component names are always available.